benchmarks/graphics_pipeline/main.cpp

```diff
@@ -285,6 +335,41 @@ void ProjApp::Setup()
         PPX_CHECKED_CALL(GetDevice()->CreateDescriptorSetLayout(&layoutCreateInfo, &mSphere.descriptorSetLayout));
     }
 
+    // Grid for sphere instances
+    {
+        mSphereGrid.xSize = static_cast<uint32_t>(std::pow(kMaxSphereInstanceCount, 1.0f / 3.0f));
+        mSphereGrid.ySize = mSphereGrid.xSize;
+        mSphereGrid.zSize = static_cast<uint32_t>(std::ceil(kMaxSphereInstanceCount / static_cast<float>(mSphereGrid.xSize * mSphereGrid.ySize)));
+        mSphereGrid.step  = 10.0f;
+    }
+
+    // Read/Generate sphere indices
+    {
+        const std::filesystem::path path = kSphereIndicesFilePath;
+        if (std::filesystem::exists(path)) {
+            mSphereIndices = ReadSphereIndicesFromJson(kSphereIndicesFilePath);
+            PPX_ASSERT_MSG(mSphereIndices.size() == kMaxSphereInstanceCount, "Size of mSphereIndices must be " << kMaxSphereInstanceCount);
+        }
+        else {
+            mSphereIndices.resize(kMaxSphereInstanceCount);
+            std::iota(mSphereIndices.begin(), mSphereIndices.end(), 0);
+            std::shuffle(mSphereIndices.begin(), mSphereIndices.end(), std::random_device{});
```

Comment:
Have you consider using deterministic pseudo random number generator? You can reproduce a sequence with a seed instead of storing the entire sequence.

Reply:
Thanks. Done.